f76/scripts/scrape/junk_locations.py

```diff
@@ -1,32 +1,159 @@
 
+import pathlib
+from bs4 import BeautifulSoup, Tag
 
-# Now I want to write a script that's a bit different
+from .infra import db_conn, fetch_soup
+from ..parsing_utils import clean_text
 
-"""
-We know that each junk item has a URL associated with it:
+BASE = "https://fallout.fandom.com"
 
-.44 Casing - URL: https://fallout.fandom.com/wiki/.44_casing
+_NUM_WORDS = {
+    "one": 1, "two": 2, "three": 3, "four": 4, "five": 5,
+    "six": 6, "seven": 7, "eight": 8, "nine": 9, "ten": 10,
+    "eleven": 11, "twelve": 12, "thirteen": 13, "fourteen": 14,
+    "fifteen": 15, "sixteen": 16, "seventeen": 17, "eighteen": 18,
+    "nineteen": 19, "twenty": 20,
+}
 
-On each of these pages there is a "Locations" section 
+def _parse_quantity(text:str) -> int | None:
+    """
+    If the description starts with a clear count word ('One', 'Two', ...),
+    return that integer, else None (e.g., 'Over twenty', 'Sometimes', etc).
+    """
+    if not text:
+        return None
+    first = text.strip().lower().split()[0]
+    return _NUM_WORDS.get(first)
 
-For this ex:
+def _first_location_link(li: Tag) -> Tag | None:
+    """
+    Heuristic: pick the first <a> in the top-level LI (not in a nested <ul>),
+    skipping image links. If none, fall back to any <a>.
+    """
+    # Top-level anchors = descendants of li but not under its first nested <ul>
+    direct_sub_ul = li.find("ul", recursive=False)
+    for a in li.find_all("a", href=True):
+        if a.find("img"): # sometimes these are imgs - skip
+            continue
+        parent_ul = a.find_parent("ul")
+        if parent_ul is direct_sub_ul:
+            # anchor is inside nested UL → skip for the "main" place
+            continue
+        return a
+    return li.find("a", href=True)
 
-- One at Harper's Ferry (<-- link on Harper's ferry)
+def _iter_sub_points(li: Tag):
+    sub_ul = li.find("ul", recursive=False)
+    if not sub_ul:
+        return
+    for sub_li in sub_ul.find_all("li", recursive=False):
+        yield sub_li
```

Comment:
I've been building a lot of adhoc functions to parse HTML markup, we should review all functions like this and see if we can't create another util module for these functions. Parsing unordered lists with sub-lists will likely be a common thing.

Reply:
To be clear, I'll do this on another PR because I actually want this functionality right now for game play.